Fixes #42: False positive in aliased custom type

gampleman opened the pull request:
The bug is basically that the code assumed that a type alias declaration creates a function with that name that would shadow any imported definition with that same name.

However, this is only the case when you declare a record type alias. Other type aliases do not affect the value level namespace at all.

jfmengels commented:
Hey :wave: 

I think you're solving a false positive, but I think it's a different one from #42, as I believe I can still reproduce the original issue.

I'll merge it in still though! Thanks a lot for the PR! :heart:

jfmengels commented:
Oh nevermind, I wrote the test that reproduced it wrong. Thanks a lot for your help @gampleman! I really appreciate you helping out here, and I'm surprised you were able to figure out how to fix the issue (it's a pretty hairy rule :/ )

gampleman commented:
Hah I blame the extensive test suite :) I just added the test, saw it fail, then searched for type alias, changed the first place that looked remotely like it might be involved, saw the test pass and submitted the PR. 

I don't pretend to actually understand any of the code 🤦
